<a href="https://colab.research.google.com/github/Mariett-7/programacion1/blob/main/parcial%202.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

ANGIE MARIETH ORTEGA JAIMES 2232446

PARCIAL PARTE 1 Y 2

PARTE 1 DEL PARCIAL

instruccion 1

In [67]:
import numpy as np
import pandas as pd

In [68]:
np.random.seed(0)
datos = np.random.randint(152, 190, 50)
datos

array([152, 155, 155, 161, 171, 173, 188, 175, 158, 176, 176, 164, 153,
       175, 176, 169, 189, 177, 165, 160, 161, 172, 168, 157, 167, 152,
       170, 187, 176, 181, 171, 171, 166, 184, 153, 161, 184, 183, 162,
       175, 187, 163, 180, 186, 152, 152, 188, 157, 169, 167])

In [69]:
np.unique(datos)


array([152, 153, 155, 157, 158, 160, 161, 162, 163, 164, 165, 166, 167,
       168, 169, 170, 171, 172, 173, 175, 176, 177, 180, 181, 183, 184,
       186, 187, 188, 189])

In [70]:
media = np.mean(datos)
media

np.float64(169.4)

In [71]:
desviación = np.std(datos)
desviación

np.float64(11.18391702401265)

In [72]:
df = pd.DataFrame(datos)
df


,0
0,152
1,155
2,155
3,161
4,171
5,173
6,188
7,175
8,158
9,176


In [73]:
df.apply(lambda x: (x - x.mean()) / x.std())

,0
0,-1.540169
1,-1.274623
2,-1.274623
3,-0.743530
4,0.141625
5,0.318656
6,1.646387
7,0.495687
8,-1.009076
9,0.584202


In [74]:
df.describe()

,0
count,50.000000
mean,169.400000
std,11.297462
min,152.000000
25%,161.000000
50%,169.500000
75%,176.000000
max,189.000000


de mis datos aleatorios se puede decir que tienen como valor minimo 152 y como valor máximo 189. y que el promedio de los datos es de 169.4 y tiene una desviación estandar de 11.3

In [85]:
from google.colab import files
import csv



In [86]:
with open("resultados.csv", "w", newline="") as f:
    writer = csv.writer(f)
    writer.writerows(df)

Error: iterable expected, not int

In [87]:
files.download("resultados.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

instrucción 2

In [35]:
datos2 = pd.DataFrame({
    "nombre":["Jacob", "Richard","Franco", "Laura", "Ana"],
    "estatura": [1.87, 1.90, 1.76, 1.62, 2],
    "nota": [5,3.4,7, 8.3, 2],
    "ciudad": ["Cali", "Cartagena", "Bogota", "Cali", "Valledupar"]
})



In [34]:
datos2.head()

,nombre,estatura,nota,ciudad
0,Jacob,1.87,5.0,Cali
1,Richard,1.90,3.4,Cartagena
2,Franco,1.76,7.0,Bogota
3,Laura,1.62,8.3,Cali
4,Ana,2.00,2.0,Valledupar


In [36]:
datos2["ciudad"]

,ciudad
0,Cali
1,Cartagena
2,Bogota
3,Cali
4,Valledupar


In [38]:
datos2.loc[datos2["nota"] >6]

,nombre,estatura,nota,ciudad
2,Franco,1.76,7.0,Bogota
3,Laura,1.62,8.3,Cali


In [41]:
datos2["nota 2"] = datos2["nota"]+1
datos2

,nombre,estatura,nota,ciudad,nota 2
0,Jacob,1.87,5.0,Cali,6.0
1,Richard,1.90,3.4,Cartagena,4.4
2,Franco,1.76,7.0,Bogota,8.0
3,Laura,1.62,8.3,Cali,9.3
4,Ana,2.00,2.0,Valledupar,3.0


In [44]:
grupo = datos2.groupby("nombre")["nota 2"].mean()
grupo

,nota 2
nombre,
Ana,3.0
Franco,8.0
Jacob,6.0
Laura,9.3
Richard,4.4


In [54]:
datos2

,nombre,estatura,nota,ciudad,nota 2
0,Jacob,1.87,5.0,Cali,6.0
1,Richard,1.90,3.4,Cartagena,4.4
2,Franco,1.76,7.0,Bogota,8.0
3,Laura,1.62,8.3,Cali,9.3
4,Ana,2.00,2.0,Valledupar,3.0


In [55]:
from google.colab import files
import csv
with open("datos_procesados.csv", "w", newline="") as f:
    writer = csv.writer(f)
    writer.writerows(datos2)
files.download("datos_procesados.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

PARTE III

In [95]:
import numpy as np
import pandas as pd


PRUEBA_A = np.random.normal(loc=7.2, scale=0.4, size=59)
PRUEBA_B = np.random.gamma(shape=306, scale=0.0229, size=59)


In [96]:
calificaciones = pd.DataFrame({
    "A": PRUEBA_A,
    "B": PRUEBA_B
})
calificaciones

,A,B
0,7.158836,6.914047
1,7.086949,7.357466
2,7.256970,6.761340
3,7.416493,6.937555
4,7.736039,7.314121
5,6.572298,7.432898
6,6.995863,6.870528
7,7.020891,5.951341
8,7.575140,6.947656
9,7.057335,7.577877


In [97]:
calificaciones.describe()

,A,B
count,59.000000,59.000000
mean,7.130323,6.968233
std,0.372849,0.375741
min,6.299386,5.951341
25%,6.833798,6.729588
50%,7.086949,6.955741
75%,7.359298,7.285663
max,8.022587,7.815262


para la prueba A, las calificaciones tienen una media de 7.130323, mientras que para la prueba B este valor es de 6.968233, lo que indica que el promedio de calificaciones es mayor en la prueba A. aún asi su desviación estandar no estan lejana siendo 0.372849 y 0.375741 para la prueba A y B respectivamente. por otro lado se tiene que la calificación minima para cada prueba fue de 6.299386 y 5.951341 lo que marca una diferencia de 0.348044 entre la nota minima de las 2 pruebas. por el momento estos datos nos muestran que si hay una diferencia promedio entre las calificaciones de ambas pruebas.

In [99]:
from scipy import stats

In [100]:
from scipy.stats import shapiro

In [102]:
shapiro(calificaciones["A"])

ShapiroResult(statistic=np.float64(0.9901498068651191), pvalue=np.float64(0.9153011232634816))

In [103]:
shapiro(calificaciones["B"])

ShapiroResult(statistic=np.float64(0.9869958783783407), pvalue=np.float64(0.7811222312486849))

como en ambas pruebas el valor p es mayor a 0.05 no se rechaza con una confianza del 95% el supuesto de normalidad

In [104]:
stats.ttest_rel

<function scipy.stats._stats_py.ttest_rel(a, b, axis=0, nan_policy='propagate', alternative='two-sided', *, keepdims=False)>

In [112]:
t_stat, p_val = stats.ttest_ind(calificaciones["A"], calificaciones["B"])
t_stat, p_val

(np.float64(2.3520712026835384), np.float64(0.02035470981758775))